### PR TITLE
First cut at a zoom-push transition.

### PR DESCRIPTION
**Pull request by jwatson**
@mattThousand early look at a zoom-push transition. Considering a pop variant that reverses the animation. Not ready for merge, but let me know what you think.

### CODE REVIEW

File: Source/Shift/UIWindow+Screenshot.swift [hunk 1]
```diff
@@ -73,7 +72,8 @@ extension UIWindow {
                 // draw view hierarchy or render
                 if window.respondsToSelector(Selector("drawViewHierarchyInRect:")) {
                     window.drawViewHierarchyInRect(window.bounds, afterScreenUpdates: true)
-                } else {
+                }
+                else {
```
Comment (mattThousand) [inline Source/Shift/UIWindow+Screenshot.swift:62]: ok :)

File: Source/Shift/ZoomPushTransition.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import UIKit
+
+
+public final class ZoomPushTransition: NSObject {
+
+    private let transitionTime = NSTimeInterval(0.35)
+    private let scaleChangePct = CGFloat(0.33)
```
Comment (mattThousand) [inline Source/Shift/ZoomPushTransition.swift]: #? thoughts about making these `public`? ++configurability

File: Source/Shift/ZoomPushTransition.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    public func transitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval {
+        return transitionTime
+    }
+
+    public func animateTransition(transitionContext: UIViewControllerContextTransitioning) {
+        guard let toVC = transitionContext.viewControllerForKey(UITransitionContextToViewControllerKey),
+              let fromVC = transitionContext.viewControllerForKey(UITransitionContextFromViewControllerKey),
+              let container = transitionContext.containerView() else {
+            debugPrint("Animation setup failed")
```
Comment (mattThousand) [inline Source/Shift/ZoomPushTransition.swift:51]: ah, `debugPrint`. The more you know... _scurries away to update other classes_

File: Source/Shift/ZoomPushTransition.swift (hunk 2 above)
Comment (jwatson) [inline Source/Shift/ZoomPushTransition.swift]: Good point, these could be `public var`.

**Comment by mattThousand** [comment]
hm, just re-reading that first comment of yours that says "not ready for merge". Well.

**Comment by mattThousand** [comment]
"I never worry about action, but only inaction." - Churchill

**Pull request merged**